### ISSUE TITLE
What does gun use as a backing store?

### ISSUE THREAD

**Issue by stuartpb** [amark/gun#177]
Does gun use IndexedDB on clients that support it? I have an application that involves a large volume of data, and I'm using PouchDB right now because localStorage-based solutions are too slow. If gun uses localStorage, does it keep documents on separate keys so as to avoid needlessly large serialization / deserialization times for large datasets?

**Comment by amark** [comment]
The default driver does not, although it shouldn't be difficult to use IndexDB. @psychollama does your Level driver plug in with IndexDB down and work out of the box?

Yes it keeps every object of every document in a separate key. Only lists that are really long are clumped together. Our 0.5.x release focuses on end-to-end performance, so don't expect much with the current version.

What are you working on? Interested in learning!

**Comment by PsychoLlama** [comment]
I'm not sure if it works with gun-level. I know I tried it at one point and ran into webpack issues with gun, but we fixed them in v0.3 and I haven't tried since... I'll give it a shot!

**Comment by PsychoLlama** [comment]
I got it compiling with webpack, and IndexedDB (though [FruitDown](https://github.com/nolanlawson/fruitdown)) seems to be working wonderfully well! There are a few things I had to do with the webpack config file, partly because I wasn't a heavy user of webpack when I built gun-level, and did a few things naively. I've put together a gist of how it all works [**here**](https://gist.github.com/PsychoLlama/5894445f48254f6f3f00).
If you have any issues or ideas, send 'em my way!
